posts/new-snapshot-format.md

```diff
@@ -0,0 +1,47 @@
+
+## Old build process overview
+
+The [`stackage`](https://github.com/commercialhaskell/stackage/) repo has a [`build-constraints.yaml`](https://github.com/commercialhaskell/stackage/blob/master/build-constraints.yaml) file, which specifies which packages should be included in the next nightly, which tests are expected to fail, upper bounds, etc. The file is where most people interact with the Stackage project, where pull requests land, and where the Stackage Curator team applies changes.
+
+The [`stackage-curator`](https://github.com/fpco/stackage-curator/) tool is used on Travis CI to put together the constraints file with the current state of Hackage to come up with a snapshot, and then check to make sure all of the bounds on that snapshot are valid. When you get an issue from someone on the Curator team reporting bounds failures, it's a report from `stackage-curator`.
+
+We run a build server that rebuilds nightly snapshots on a regular basis. It uses the `stackage-curator` tool as well to create the snapshot definition, perform the build, run tests, upload Haddocks to Amazon S3, and commit the new snapshot file to the [`stackage-nightly`](https://github.com/commercialhaskell/stackage-nightly/) repo. If there are build failures, the Stackage Curator team reports on them and adds more constraints to work around them.
+
+LTS snapshots are basically the same thing, except for an LTS minor version bump, the constraints come from the previous snapshot (e.g., `lts-13.22` is based on the constraints implicit in `lts-13.21`), and the snapshots are committed to the [`lts-haskell`](https://github.com/commercialhaskell/lts-haskell/) repo.
+
+Finally, [Stackage Server](https://github.com/fpco/stackage-server/) (a.k.a. [stackage.org](https://www.stackage.org/)) takes information from `stackage-nightly`, `lts-haskell`, and the uploaded docs on S3 to provide everything it does: lists of snapshots, packages, Haddocks, and Hoogle search.
+
+## Problems
+
+* `stackage-curator` implements its own Cabal package building logic by working directly with Cabal-the-library. There's a lot of feature overlap with Stack, and different behavior in some corner cases.
+* The original Stackage file format used in `stackage-nightly` and `lts-haskell` is unique from the "custom snapshot" format that Stack supports, for only historical reasons.
+* The snapshot format is _big_. It contains a lot of information about module names, constraints, users, etc, that build tools don't need.
+* While there is cryptographic hash information on cabal files to deal with Hackage Revisions, there is no cryptographic hash information on the package tarball contents themselves.
+
+## Changes
+
+* We're scrapping the previous `stackage-curator` tool entirely.
+* There's a brand new `curator` tool, which uses the same snapshot format as Stack 2, Pantry, and what was previously known as custom snapshots.
```

Comment:
Does the new `curator` tool have a repo? Might be worth including the link to it, IMO.

Reply:
Doh, good catch! It's in the Stack repo, I'll link to the subdir.